lib/rmt/cli/products.rb

```diff
@@ -2,28 +2,36 @@
 
 class RMT::CLI::Products < RMT::CLI::Base
 
+  include ::RMT::CLI::ArrayPrintable
+
   desc 'list', 'List products which are marked to be mirrored.'
   option :all, aliases: '-a', type: :boolean, desc: 'List all products, including ones which are not marked to be mirrored'
   option :release_stage, aliases: '-r', type: :string, desc: 'beta, released'
-  def list
-    attributes = %i[id name version arch product_string release_stage mirror? last_mirrored_at]
-    headings = ['ID', 'Name', 'Version', 'Architecture', 'Product string', 'Release stage', 'Mirror?', 'Last mirrored']
+  option :csv, type: :boolean, desc: 'Output data in CSV format'
 
+  def list
     products = options.all ? Product.all : Product.mirrored
-    rows = products.with_release_stage(options[:release_stage]).map do |product|
-      attributes.map { |a| product.public_send(a) }
-    end
+    products = products.with_release_stage(options[:release_stage])
 
-    if rows.empty?
+    if products.empty?
       if options.all
         warn 'Run "rmt-cli sync" to synchronize with your SUSE Customer Center data first.'
       else
         warn 'No matching products found in the database.'
       end
     else
-      puts Terminal::Table.new(headings: headings, rows: rows)
+      puts format_array(products, {
+        id: 'ID',
+        name: 'Name',
+        version: 'Version',
+        arch: 'Architecture',
+        product_string: 'Product string',
+        release_stage: 'Release stage',
+        'mirror?' => 'Mirror?',
+        last_mirrored_at: 'Last mirrored'
+      })
```

Comment:
I like the explicitness of this, The previous `public_send` thing was not nice. :+1: